packages/premint-sdk/src/premint-api.ts

```diff
@@ -0,0 +1,684 @@
+   * @returns isValid = signature is valid or not, contractAddress = assumed contract address, recoveredSigner = signer from contract
+   */
+  async isValidSignature({
+    data,
+    publicClient,
+  }: {
+    data: PremintResponse;
+    publicClient?: PublicClient;
+  }): Promise<{
+    isValid: boolean;
+    contractAddress: Address;
+    recoveredSigner: Address;
+  }> {
+    publicClient = this.getPublicClient(publicClient);
+
+    const [isValid, contractAddress, recoveredSigner] =
+      await publicClient.readContract({
+        abi: zoraCreator1155PremintExecutorImplABI,
+        address: this.getExecutorAddress(),
+        functionName: "isValidSignature",
+        args: [data.collection, convertPremint(data.premint), data.signature],
+      });
+
+    return { isValid, contractAddress, recoveredSigner };
+  }
+
+  /**
+   * Execute premint on-chain
+   *
+   * @param settings.data Data from the API for the mint
+   * @param settings.account Optional account (if omitted taken from wallet client) for the account executing the premint.
+   * @param settings.walletClient WalletClient to send execution from.
+   * @param settings.mintArguments User minting arguments.
+   * @param settings.mintArguments.quantityToMint Quantity to mint, optional, defaults to 1.
+   * @param settings.mintArguments.mintComment Optional mint comment, optional, omits when not included.
+   * @param settings.publicClient Optional public client for preflight checks.
+   * @returns receipt, log, zoraURL
+   */
+  async executePremintWithWallet({
+    data,
+    account,
+    walletClient,
+    mintArguments,
+    publicClient,
+  }: {
+    data: PremintResponse;
+    walletClient: WalletClient;
+    account?: Account | Address;
+    mintArguments?: {
+      quantityToMint: number;
+      mintComment?: string;
+    };
+    publicClient?: PublicClient;
+  }) {
+    publicClient = this.getPublicClient(publicClient);
+
+    if (mintArguments && mintArguments?.quantityToMint < 1) {
+      throw new Error("Quantity to mint cannot be below 1");
+    }
+
+    const targetAddress = this.getExecutorAddress();
+    const numberToMint = BigInt(mintArguments?.quantityToMint || 1);
+    const args = [
+      data.collection,
+      convertPremint(data.premint),
+      data.signature,
+      numberToMint,
+      mintArguments?.mintComment || "",
+    ] as const;
+
+    if (!account) {
+      account = walletClient.account;
+    }
+
+    if (!account) {
+      throw new Error("Wallet not passed in");
+    }
+
+    const value = numberToMint * this.rewardPerToken;
+
+    const { request } = await publicClient.simulateContract({
+      account,
+      abi: zoraCreator1155PremintExecutorImplABI,
+      functionName: "premint",
+      value,
+      address: targetAddress,
+      args,
+    });
+    const hash = await walletClient.writeContract(request);
+    const receipt = await publicClient.waitForTransactionReceipt({ hash });
+    const log = getLogFromReceipt(receipt);
```

Comment:
as mentioned above, reading this code and looking at the api definition its not really clear why you would need this `log`

the main reason to need it is to get the `tokenId` - so imo it would be clearer if you just extract the `tokenId` from the logs and return it

Reply:
actually I'd lean towards just returning the raw transaction receipt, and then having utility function to parse it like:

`getTokenIdFromReceipt(receipt)`

or

`getCreatorAttributionFromReceipt(receipt)`

the way the `log` is parsed now is opinionated and assumes a desired set of data, so returning the raw receipt and allowing the end user to extract the data they need with helper functions from it would add more flexibility

Reply:
We return both – having users parse logs is annoying and lots of users ask how to best do it. I don't mint having helper functions but assume most users just want these two fields + urls. The raw txn receipt is already returned.